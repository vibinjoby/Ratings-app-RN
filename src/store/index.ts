import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { apiCallBegan } from './actions/api'

import api from './middleware/api'
import rootReducer from './reducers/rootReducer'

export const store = configureStore({
  reducer: rootReducer,
  middleware: [
    ...getDefaultMiddleware({
      serializableCheck: {
        // Ignore these action types
        ignoredActions: [apiCallBegan.type],
      },
    }),
    api,
  ],
})

// Infer the `RootState` from the store itself
export type RootState = ReturnType<typeof store.getState>
