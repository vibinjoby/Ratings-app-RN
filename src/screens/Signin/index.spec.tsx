import 'react-native'
import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'

import SignIn from '.'
import { Provider } from 'react-redux'
import { store } from '../../store'

describe('Sign In screen', () => {
  it('should render', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <Provider store={store}>
          <SignIn />
        </Provider>
      </NavigationContainer>,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should let me enter values', () => {
    const { getByTestId } = render(
      <NavigationContainer>
        <Provider store={store}>
          <SignIn />
        </Provider>
      </NavigationContainer>,
    )
    fireEvent.changeText(getByTestId('email'), 'Email')
    fireEvent.changeText(getByTestId('password'), 'Password')
    fireEvent.press(getByTestId('signinBtn'))
  })
})
