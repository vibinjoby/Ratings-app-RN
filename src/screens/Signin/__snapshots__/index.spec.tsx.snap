// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In screen should render 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "height": 240,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/authBg/authBg.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 240,
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loginDish/loginDish.png",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#181725",
              "fontFamily": "Poppins-Regular",
              "fontSize": 26,
              "lineHeight": 29,
            }
          }
        >
          Log In
        </Text>
        <View
          style={
            Object {
              "alignSelf": "center",
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomLeftRadius": 40,
                      "borderTopLeftRadius": 40,
                      "marginRight": 2,
                    },
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#E2E2E2",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "elevation": 3,
                      "paddingHorizontal": 14,
                      "paddingVertical": 8,
                      "shadowColor": "#181725",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 0.22,
                      "shadowRadius": 2.22,
                    },
                    Object {
                      "backgroundColor": "#FF5823",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins-Regular",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#fff",
                      },
                    ]
                  }
                >
                  Customer
                </Text>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 2,
                    },
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#E2E2E2",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "elevation": 3,
                      "paddingHorizontal": 14,
                      "paddingVertical": 8,
                      "shadowColor": "#181725",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 0.22,
                      "shadowRadius": 2.22,
                    },
                    false,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins-Regular",
                        "fontSize": 16,
                      },
                      false,
                    ]
                  }
                >
                  Owner
                </Text>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomRightRadius": 40,
                      "borderTopRightRadius": 40,
                    },
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#E2E2E2",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "elevation": 3,
                      "paddingHorizontal": 14,
                      "paddingVertical": 8,
                      "shadowColor": "#181725",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 0.22,
                      "shadowRadius": 2.22,
                    },
                    false,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins-Regular",
                        "fontSize": 16,
                      },
                      false,
                    ]
                  }
                >
                  Admin
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 30,
              "marginTop": 40,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#7C7C7C",
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
              }
            }
          >
            Email/Username
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#E2E2E2",
                  "color": "#181725",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                  "paddingVertical": 10,
                  "width": "100%",
                }
              }
              testID="email"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#7C7C7C",
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
              }
            }
          >
            Password
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#E2E2E2",
                  "color": "#181725",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                  "paddingVertical": 10,
                  "width": "100%",
                }
              }
              testID="password"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                }
              }
              testID="eyeIc"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/password/password.png",
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E2E2E2",
              "borderRadius": 19,
              "marginTop": 40,
              "opacity": 1,
              "padding": 24,
            }
          }
          testID="signinBtn"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                },
                Object {},
              ]
            }
          >
            Log In
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 25,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Poppins-Regular",
                "fontSize": 14,
              }
            }
          >
            Don't have an account? 
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#54BF29",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 14,
                }
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
